react-components/stories/SearchHooks.stories.tsx

```diff
@@ -0,0 +1,292 @@
+            )
+          );
+
+          return isInExternalId || isInProperties;
+        }) ?? []
+      );
+    } else if (searchMethod === 'allAssets') {
+      const transformedAssets =
+        allAssets?.pages
+          .flat()
+          .map((mapping) => mapping.assets)
+          .flat() ?? [];
+
+      const filteredAssets =
+        transformedAssets.filter((asset) => {
+          const isInName = asset.name.toLowerCase().includes(mainSearchQuery.toLowerCase());
+          const isInDescription = asset.description
+            ?.toLowerCase()
+            .includes(mainSearchQuery.toLowerCase());
+
+          return isInName || isInDescription;
+        }) ?? [];
+
+      const mappedAssets: Equipment[] = filteredAssets.map((asset) => {
+        return {
+          view: 'Asset',
+          externalId: asset.id + '',
+          space: 'Whole project',
+          properties: {
+            name: asset.name,
+            description: asset.description
+          }
+        };
+      });
+
+      return mappedAssets;
+    } else if (searchMethod === 'assetSearch') {
+      if (assetSearchData === undefined) {
+        return [];
+      }
+
+      const searchedEquipment: Equipment[] = assetSearchData.map((asset) => {
+        return {
+          view: 'Asset',
+          externalId: asset.id + '',
+          space: 'Whole project',
+          properties: {
+            name: asset.name,
+            description: asset.description
+          }
+        };
+      });
+
+      return searchedEquipment;
+    } else if (searchMethod === 'fdmSearch') {
+      if (searchData === undefined) {
+        return [];
+      }
+
+      const searchedEquipment: Equipment[] = searchData
+        .map((searchResult) => {
+          return searchResult.instances.map((instance) => {
+            return {
+              view: searchResult.view.externalId,
+              externalId: instance.externalId,
+              space: instance.space,
+              properties: instance.properties
+            };
+          });
+        })
+        .flat();
+
+      return searchedEquipment;
+    } else {
+      return [];
+    }
+  }, [mainSearchQuery, allEquipment, searchData, allAssets, assetSearchData, searchMethod]);
+
+  return (
+    <>
+      <RevealContainer sdk={sdk} color={new Color(0x4a4a4a)}>
+        <ReactQueryDevtools position="bottom-right" />
+        <RevealResourcesFitCameraOnLoad
+          resources={resources}
+          defaultResourceStyling={{
+            cad: {
+              default: { color: new Color('#efefef') },
+              mapped: { color: new Color('#c5cbff') }
+            }
+          }}
+        />
+        <RevealToolbar />
+      </RevealContainer>
+      <h1>Mapped equipment</h1>
+      <div style={{ display: 'flex', flexDirection: 'row', gap: 16, padding: '0 8px 8px 0' }}>
+        <input
+          onInput={(event) => {
+            setTempSearchQuery((event.target as HTMLInputElement).value);
+          }}></input>
```

Comment:
It's very nice to see all the query functions in action! Can we convert these buttons to radio-buttons instead? Or perhaps just a label / header showing the currently selected search mode?

Reply:
Styled it with Cogs so it looks fancy now 💅
![image](https://github.com/cognitedata/reveal/assets/45129444/23d959c5-a25c-443d-b4f7-446d6f09ef53)